### PR TITLE
Feature #413: container ephemeral storage request and limit check

### PR DESCRIPTION
**Pull request by kmarteaux**
```
RELNOTE: Implement new Pod container resource checks for ephemeral storage. Default tests for request and limit settings. Optional test to ensure matching ephemeral storage request and limit values.
```

### CODE REVIEW

**Comment by kmarteaux** [comment]
Refactored the original implementation - moved the new checks to score/container.go and the associated tests to score/score_test.go.  Split the resource checks into default (are they set) and optional (are they equal) as previously recommended. Removed the separate score/ephemeralstorage/ephemeralstorage.go and score/ephemeralstorage_test.go files. Updated score.go to remove ref to deleted files.

File: score/score_test.go [hunk 1]
```diff
@@ -307,3 +306,37 @@ func TestList(t *testing.T) {
...
 }
+
+func TestPodContainerStorageEphemeralNoLimit(t *testing.T) {
+	t.Parallel()
+	testExpectedScore(t, "pod-ephemeral-storage-missing-limit.yaml", "Container Ephemeral Storage Request and Limit", scorecard.GradeCritical)
+}
+
+func TestPodContainerStorageEphemeralNoRequest(t *testing.T) {
+	t.Parallel()
+	testExpectedScore(t, "pod-ephemeral-storage-missing-request.yaml", "Container Ephemeral Storage Request and Limit", scorecard.GradeWarning)
```
Comment (zegl) [inline score/score_test.go:317]: Could we have a test that checks for "Container Ephemeral Storage Request and Limit" returning GradeOK?

**Comment by zegl** [comment]
Thanks! Looks great 👍 , there's just one test case (the happy path) that isn't tested. Update the PR with that, and we're ready to ship!

File: score/score_test.go (hunk 1 above)
Comment (kmarteaux) [reply score/score_test.go:324]: Additional test case (GradeAllOK) added.

**Review by zegl — APPROVED** [review]
Great!

bors merge

**Pull request merged**